src/katsdpcontroller/product_config.py

```diff
@@ -1396,6 +1397,81 @@ def from_config(
         )
 
 
+class GpucbfTiedArrayResampledVoltageStream:
+    """Real tied-array-resampled-voltage stream (GPU V-engine)."""
+
+    stream_type: ClassVar[str] = "gpucbf.tied_array_resampled_voltage"
+    _valid_src_types: ClassVar[_ValidTypes] = {"gpucbf.tied_array_channelised_voltage"}
+
+    def __init__(
+        self,
+        name: str,
+        src_streams: Sequence[Stream],
+        *,
+        n_chans: int,
+        pols: Sequence[str],
+        station_id: str,
+    ):
+        if len(src_streams) != 2:
+            raise ValueError(f"{self.stream_type} requires two {self._valid_src_types}")
+        grandparent_streams: List[GpucbfAntennaChannelisedVoltageStream] = [
+            tacv_src_streams.src_streams[0]
+            for tacv_src_streams in src_streams
+            if len(tacv_src_streams.src_streams) == 1
+            and isinstance(tacv_src_streams.src_streams[0], GpucbfAntennaChannelisedVoltageStream)
```

Comment:
```suggestion
```
These conditions are already enforced by the combination of the JSON schema and `_valid_src_types`.

Reply:
Noted all around. I'm struggling to convince mypy that I know what I'm doing with `grandparent_streams` without this `isinstance` check. I've updated to read
```python
grandparent_streams = [
            tacv_src_streams.src_streams[0] for tacv_src_streams in src_streams
        ]
assert all(isinstance(stream, GpucbfAntennaChannelisedVoltageStream) for stream in grandparent_streams)
```

But it still doesn't acknowledge the "updated type check/confirmation" after the assertion. Any suggestions?